src/ray/gcs/redis_context.h

```diff
@@ -38,6 +42,20 @@ namespace gcs {
 
 using rpc::TablePrefix;
 
+template <typename RedisContextType>
+struct RedisDeleter {
+  RedisDeleter(){};
+
+  void operator()(RedisContextType *context) {}
+};
+
+template <>
+struct RedisDeleter<redisContext> {
```

Comment:
Don't we need RedisDeleter for async context?

Reply:
The existing (FreeRedisContext) deleter was an empty deleter, so the non-specialized template should work for that type?

Reply:
Hmm but we *do* need this to the unique_ptr comment above

Reply:
The reason it was an empty deleter for redisAsyncContext was because somebody put it directly into the destructor of the RedisAsyncContext wrapper class, lol

Reply:
Yea. I'm saying for the consistency we don't let destructor of the RedisAsyncContext delete the context but also let unique_ptr deleter to delete it. https://github.com/ray-project/ray/pull/41771#discussion_r1421876180

Reply:
Yeah, agree